### PR TITLE
5023 timeline table steps styling

### PR DESCRIPTION
**Pull request by emitche**
- Style steps in timeline table
- Add opacity to project step colors

### CODE REVIEW

**Comment by emitche** [comment]
image

This is the color difference with steps. This is an example of a project without groups with several completed steps (early, on time, late, very late, super late).

**Comment by emitche** [comment]
Spacers were changed to white, per @rgalusca's request in this branch. Spacer styling is part of Madeline 5158.

image

File: app/models/project_step.rb [hunk 1]
```diff
@@ -221,34 +221,38 @@ def old_start_date_statement
   end
 
   # Generate a CSS color based on the status and lateness of the step
-  def color
+  def color(opacity)
```
Comment (hooverlunch) [inline app/models/project_step.rb]: I suggest opacity should have a default value of 1 here. Then you only need pass it if you want something that is not opaque.

File: app/models/project_step.rb [hunk 2]
```diff
@@ -359,7 +363,7 @@ def handle_scheduled_start_date
   end
 
   # start and finish are each CSS color strings in hsl format
-  def color_between(start, finish, fraction = 0.5)
+  def color_between(start, finish, fraction = 0.5, opacity)
```
Comment (hooverlunch) [inline app/models/project_step.rb]: Opacity default here as well?

File: app/views/admin/loans/timeline/_step.slim [hunk 3]
```diff
@@ -1,15 +1,41 @@
...
 *wrapper_tag
   / Add spacer for step who has a group as a sibling
   - unless siblings_are_all_steps
     td.spacer colspan="#{max_group_depth - entry.depth + 1}"
-  td = entry.summary
-  td = entry.display_start_date ? l(entry.display_start_date, format: :short) : ''
-  td = entry.display_end_date ? l(entry.display_end_date, format: :short) : ''
-  td
+
+  td class="#{classes}"
```
Comment (hooverlunch) [inline app/views/admin/loans/timeline/_step.slim]: I think you can do just `class=classes`, without the quotes or `#{}`. Give it a try?

Comment (hooverlunch) [inline app/views/admin/loans/timeline/_step.slim]: Also, can you put the class on the tr/div and then do `tr.whatever td` in your stylesheet? This is what I usually do for tables and it saves a lot of duplication.

File: app/views/admin/loans/timeline/_step.slim [hunk 4]
```diff
@@ -1,15 +1,41 @@
...
+    input.select-step type="checkbox" class=(entry.completed? ? "completed-item" : "incomplete-item") data-id=entry.id
```
Comment (hooverlunch) [inline app/views/admin/loans/timeline/_step.slim]: Could you reuse the `completed` variable here?

Comment (emitche) [inline app/views/admin/loans/timeline/_step.slim]: I'm not sure why we have this here unless it is here for when we select all complete/incomplete items. But that would not be the best way to go about selecting all completed, given that completed class will be in the parent element. Just something to consider.

**Comment by emitche** [comment]
Changes made.

**Pull request merged**